dapps/assets/src/components/contacts/Contacts.vue

```diff
@@ -0,0 +1,175 @@
+          ></evan-button>
+          <evan-button
+            @click="filterByType('company')"
+            class="filter-btn ml-3"
+            type="text-secondary"
+            icon="mdi mdi-domain"
+            iconPosition="left"
+            :class="{ 'active': filter === 'company' }"
+            :label="$t('_assets.contacts.companies')"
+          ></evan-button>
+          <evan-button
+            @click="filterByType('iot-device')"
+            class="filter-btn ml-3"
+            type="text-secondary"
+            icon="mdi mdi-radio-tower"
+            iconPosition="left"
+            :class="{ 'active': filter === 'iot-device' }"
+            :label="$t('_assets.contacts.iot-devices')"
+          ></evan-button>
+          <evan-button
+            @click="resetFilter"
+            class="filter-btn ml-3"
+            type="text-secondary"
+            icon="mdi mdi-account-multiple-outline"
+            iconPosition="left"
+            :class="{ 'active': filter === null }"
+            :label="$t('_assets.contacts.all')"
+          ></evan-button>
+        </div>
+      </div>
+
+      <div class="d-flex flex-row mt-3">
+        <b-table
+          class="evan-table"
+          hover
+          :items="data"
+          :fields="columns"
+          :filter="filter"
+          :filterIncludedFields="filterBy"
+          :tbody-tr-class="'evan-table-body-row'"
+          :thead-tr-class="'evan-table-head-row'"
+          :thead-class="'evan-table-head'"
+          @row-clicked="handleRowClicked"
+          sticky-header="80vh"
+          show-empty
+        >
+          <template
+            v-slot:cell(alias)="data"
+          >{{ data.item.alias ? data.item.alias: data.item.address }}</template>
+          <template v-slot:cell(icon)="data">
+            <i class="table-icon" :class="data.item.icon"></i>
+          </template>
+          <template v-slot:cell(favorite)="data">
+            <i class="table-icon" :class="{'mdi mdi-star': data.item.favorite}"></i>
+          </template>
+
+          <!-- Empty slots -->
+          <template v-slot:empty>
+            <h4>{{ '_assets.contacts.contacts-empty' | translate }}</h4>
+          </template>
+          <template v-slot:emptyfiltered>
+            <h4>{{ '_assets.contacts.filtered-empty' | translate }}</h4>
+          </template>
+        </b-table>
+      </div>
+      <evan-loading v-if="isLoading" :classes="'mt-3'"></evan-loading>
+    </div>
+
+    <evan-button
+      :type="'icon-primary'"
+      size="lg"
+      class="add-contact-btn"
+      icon="mdi mdi-plus"
+      @click="$refs.addContact.showPanel()"
+    ></evan-button>
+
+    <add-contact ref="addContact" @contact-added="handleContactAdded" />
+  </div>
+</template>
+
+<script lang="ts">
+import ContactsComponent from './Contacts';
+export default ContactsComponent;
+</script>
+
+<style lang="scss" scoped>
+@import '~@evan.network/ui/src/style/utils';
+
+.content {
```

Comment:
Already saw this one within the profile. We should add a core class for this.

Reply:
also I'm not happy with inline stylings, would it be possible to commit to put stylings always in *scss files?

Reply:
We should discuss this in general, because I'm actually a fan of keeping all template related things in one file and not pollute the folder structure with so many partials. Should do this during retro or HQOTD